// react-vis/es/legends/discrete-color-legend.js
var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

// Copyright (c) 2016 - 2017 Uber Technologies, Inc.
//
// Permission is hereby granted, free of charge, to any person obtaining a copy
// of this software and associated documentation files (the "Software"), to deal
// in the Software without restriction, including without limitation the rights
// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
// copies of the Software, and to permit persons to whom the Software is
// furnished to do so, subject to the following conditions:
//
// The above copyright notice and this permission notice shall be included in
// all copies or substantial portions of the Software.
//
// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
// THE SOFTWARE.

import React from 'react';

import PropTypes from 'prop-types';

import DiscreteColorLegendItem from './discrete-color-legend-item';
import { DISCRETE_COLOR_RANGE } from '../theme';

var propTypes = {
  className: PropTypes.string,
  items: PropTypes.arrayOf(PropTypes.oneOfType([PropTypes.shape({
    title: PropTypes.string.isRequired,
    color: PropTypes.string,
    disabled: PropTypes.bool
  }), PropTypes.string.isRequired])).isRequired,
  onItemClick: PropTypes.func,
  height: PropTypes.number,
  width: PropTypes.number,
  orientation: PropTypes.oneOf(['vertical', 'horizontal'])
};

var defaultProps = {
  className: '',
  colors: DISCRETE_COLOR_RANGE,
  orientation: 'vertical'
};

/**
 * Add color information to a list of legend items.
 * @param {Array} items - list of legend items
 * @param {string} [items.title] - label of the legend item
 * @param {string} [items.color] - color of the legend item
 * @param {Array} colors - array of colors.
 * @returns {Array} Array of tick values.
 */

function fillItemsWithDefaults(items, colors) {
  return items.map(function (item, i) {
    return {
      title: item.title ? item.title : item,
      color: item.color ? item.color : colors[i % colors.length],
      disabled: Boolean(item.disabled)
    };
  });
}

function DiscreteColorLegend(_ref) {
  var colors = _ref.colors,
      initialItems = _ref.items,
      width = _ref.width,
      height = _ref.height,
      onItemClick = _ref.onItemClick,
      orientation = _ref.orientation,
      className = _ref.className;

  var updatedItems = fillItemsWithDefaults(initialItems, colors);
  return React.createElement(
    'div',
    {
      className: 'rv-discrete-color-legend ' + orientation + ' ' + className,
      style: { width: width, height: height } },
    updatedItems.map(function (item, i) {
      return React.createElement(DiscreteColorLegendItem, _extends({}, item, {
        orientation: orientation,
        key: i,
        onClick: onItemClick ? function () {
          return onItemClick(initialItems[i], i);
        } : null }));
    })
  );
}

DiscreteColorLegend.displayName = 'DiscreteColorLegendItem';
DiscreteColorLegend.propTypes = propTypes;
DiscreteColorLegend.defaultProps = defaultProps;

export default DiscreteColorLegend;
